tests_integration/utils.py

```diff
@@ -130,6 +130,25 @@ def kubectl_wait_pod_running(pod_name: str, max_retries: int = 10) -> dict:
     return kubectl_wait_pod_status(pod_name, "Running", max_retries=max_retries)
 
 
+def kubectl_wait_deployment_available(
+    deployment_name: str, *, max_retries: int = 5
+) -> dict:
+    retry = 0
+    while True:
+        try:
+            result = kubectl(
+                f"wait --for=condition=available --timeout=60s deployment/{deployment_name} -o json"
```

Comment:
Can we set the `timeout=max_retries * 60` (ignoring the sleep in between iterations) and remove the loop? In the worse case, we have to wait the same amount of times.

Reply:
only if the issue is timeout.
if there's a different issue - for example, deployment doesnt exist yet - it'll fail immediately and not retry